control-plane/csi-driver/src/context.rs

```diff
@@ -0,0 +1,162 @@
+    ctrl_loss_tmo: Option<u32>,
+    fs_type: Option<FileSystem>,
+}
+impl PublishParams {
+    /// Get the `Parameters::IoTimeout` value.
+    pub fn io_timeout(&self) -> &Option<u32> {
+        &self.io_timeout
+    }
+    /// Get the `Parameters::CtrlLossTmo` value.
+    pub fn ctrl_loss_tmo(&self) -> &Option<u32> {
+        &self.ctrl_loss_tmo
+    }
+}
+impl TryFrom<&HashMap<String, String>> for PublishParams {
+    type Error = tonic::Status;
+
+    fn try_from(args: &HashMap<String, String>) -> Result<Self, Self::Error> {
+        let fs_type = match args.get(Parameters::FileSystem.as_ref()) {
+            Some(fs) => FileSystem::from_str(fs.as_str())
+                .map(Some)
+                .map_err(|_| tonic::Status::invalid_argument("Invalid filesystem type"))?,
+            None => None,
+        };
+
+        let io_timeout = Parameters::io_timeout(args.get(Parameters::IoTimeout.as_ref()))
+            .map_err(|_| tonic::Status::invalid_argument("Invalid I/O timeout"))?;
+        let ctrl_loss_tmo =
+            Parameters::ctrl_loss_tmo(args.get(Parameters::NvmeCtrlLossTmo.as_ref()))
+                .map_err(|_| tonic::Status::invalid_argument("Invalid ctrl_loss_tmo"))?;
+
+        Ok(Self {
+            io_timeout,
+            ctrl_loss_tmo,
+            fs_type,
+        })
+    }
+}
+
+/// Volume Creation parameters.
+#[allow(dead_code)]
+pub struct CreateParams {
+    io_timeout: Option<u32>,
+    ctrl_loss_tmo: Option<u32>,
+    fs_type: Option<FileSystem>,
+    share_protocol: VolumeShareProtocol,
+    replica_count: u8,
+}
+impl CreateParams {
+    /// Get the `Parameters::ShareProtocol` value.
+    pub fn share_protocol(&self) -> VolumeShareProtocol {
+        self.share_protocol
+    }
+    /// Get the `Parameters::ReplicaCount` value.
+    pub fn replica_count(&self) -> u8 {
+        self.replica_count
+    }
+}
+impl TryFrom<&HashMap<String, String>> for CreateParams {
+    type Error = tonic::Status;
+
+    fn try_from(args: &HashMap<String, String>) -> Result<Self, Self::Error> {
+        let fs_type = match args.get(Parameters::FileSystem.as_ref()) {
+            Some(fs) => FileSystem::from_str(fs.as_str())
+                .map(Some)
+                .map_err(|_| tonic::Status::invalid_argument("Invalid filesystem type"))?,
+            None => None,
+        };
+
+        // Check storage protocol.
+        let share_protocol = parse_protocol(args.get(Parameters::ShareProtocol.as_ref()))?;
+
+        let io_timeout = Parameters::io_timeout(args.get(Parameters::IoTimeout.as_ref()))
+            .map_err(|_| tonic::Status::invalid_argument("Invalid I/O timeout"))?;
+        let ctrl_loss_tmo =
+            Parameters::ctrl_loss_tmo(args.get(Parameters::NvmeCtrlLossTmo.as_ref()))
+                .map_err(|_| tonic::Status::invalid_argument("Invalid ctrl_loss_tmo"))?;
+
+        let replica_count = match args.get(Parameters::ReplicaCount.as_ref()) {
+            Some(c) => match c.parse::<u8>() {
```

Comment:
Does this mean it will fail with >= 256 replicas?

Reply:
Yes correct, I think the core also uses u8 for replica count.